src/systems/spacecraft_thruster_model/SpacecraftThrusterModel.hh

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2019 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef GZ_SIM_SYSTEMS_SPACECRAFTTHRUSTERMODEL_HH_
+#define GZ_SIM_SYSTEMS_SPACECRAFTTHRUSTERMODEL_HH_
+
+#include <gz/sim/System.hh>
+#include <memory>
+
+namespace gz
+{
+namespace sim
+{
+// Inline bracket to help doxygen filtering.
+inline namespace GZ_SIM_VERSION_NAMESPACE {
+namespace systems
+{
+  // Forward declaration
+  class SpacecraftThrusterModelPrivate;
+
+  /// \brief This system applies a thrust force to models with spinning
+  /// propellers. See examples/worlds/quadcopter.sdf for a demonstration.
```

Comment:
I didn't know spacecrafts had propellers :upside_down_face: :rofl: 

On a more serious note its worth pointing users to a citation that describes the dynamics of the plugin.
Also, we usually describe the parameters on most plugins in this section.

See: https://github.com/gazebosim/gz-sim/blob/9e60fdc5dea81344d6b3c9c715ef27f1fc21968a/src/systems/hydrodynamics/Hydrodynamics.hh#L49
for an example.

Also an example would help greatly. I know you've written a tutorial

Reply:
I may or may not have templated from MulticopterMotorModel :eyes:  

Thanks for the insightful comments, I'll patch it up!